src/components/course/data/hooks.jsx

```diff
@@ -407,3 +409,60 @@ export const useTrackSearchConversionClickHandler = ({ href, eventName }) => {
 
   return handleClick;
 };
+
+/**
+ * A React hook to render an alert when a learner fails to enroll in a course for any number of
+ * reasons. The contents of the alert are determined by a ``failureReasonSlug`` which is passed
+ * from the Data Sharing Consent (DSC) page.
+ *
+ * @param {object} args
+ * @param {object} args.enterpriseConfig Details about the learner's enterprise customer.
+ * @param {boolean} args.isEnrollmentFailed Whether learner's enrollment failed due to an error
+ * @param {boolean} args.failureReasonSlug The lookup key for the reason why the enrollment
+ *  failed; used to determine the appropriate messaging to show in the alert.
+ *
+ * @returns Function to render the alert.
+ */
+export const useRenderFailedEnrollmentAlert = ({
```

Comment:
is there a reason we don't have this as another component?

Reply:
Honestly, no particular reason lol. I think I was originally going to have the custom hook return the failure message, but ultimately decided for it to return component instead. I agree it makes sense as it's own component, though. Will update!